clients/roscpp/src/libros/callback_queue.cpp

```diff
@@ -32,11 +32,6 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-// Make sure we use CLOCK_MONOTONIC for the condition variable wait_for if not Apple.
-#ifndef __APPLE__
-#define BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC
-#endif
-
```

Comment:
Your patch works fine for me even without #1651.

But why was this section removed independently of https://github.com/ros/ros_comm/pull/1651/files#diff-5f391de72dc7aa5abcad51904f99a29a? With Boost <1.61 (e.g. in Ubuntu Xenial) the preprocessor check in [callback_queue.cpp:43](https://github.com/ros/ros_comm/pull/1684/files#diff-5f391de72dc7aa5abcad51904f99a29aL48) triggers an error now. For all Boost versions <1.67 condition variables use the non-monotonic system clock by default, with undesired behavior in case of jumps in system time, but `condition_variable::wait_for` and `condition_variable::wait_until` will still do the right thing in both cases.

Reply:
Removing this lines makes the tests pass. The problem is, that with this lines the definition of condition_variable is differente in callback_queue and in the tests. This ORD-violation leads to test failures.